libcudacxx/docs/extended_api/synchronization_primitives/barrier.md

```diff
@@ -60,6 +61,58 @@ Programs shall ensure that this transformation would not introduce errors, for
 Under CUDA Compute Capability 6 (Pascal) or prior, an object of type
   `cuda::barrier` or `cuda::std::barrier` may not be used.
 
+## Shared memory barriers with transaction count
+
+In addition to the arrival count, a `cuda::barrier<thread_scope_block>` object
+located in shared memory supports a
+[tx-count](https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#tracking-asynchronous-operations-by-the-mbarrier-object),
+which is used for tracking the completion of some asynchronous memory operations or
+transactions.
+The tx-count tracks the number of asynchronous transactions, in
+units specified by the asynchronous memory operation (typically bytes), that are
+outstanding and yet to be complete.
+This capability is exposed, starting with the Hopper architecture (CUDA Compute
+Capability 9).
+
+The tx-count of `cuda::barrier` must be set to the total amount of asynchronous
+memory operations, in units as specified by the asynchronous operations, to be
+tracked by the current phase. This can be achieved with the
+`cuda::device::barrier_arrive_tx` function call. Upon completion of each of the
+asynchronous operations, the tx-count of the `cuda::barrier` will be updated and
+thus progress the `cuda::barrier` towards the completion of the current phase.
+This may complete the current phase.
+
+### Phase Completion of a `cuda::barrier` with tx-count support
```

Comment:
Btw, lines 19-20 of this file (https://github.com/NVIDIA/cccl/pull/358/files#diff-2f7a92f8ce801caa513c4c823f3982c625d0815166ee448df9788b6f937a46a1R19-R20) currently contradict this.. 

> It has the same interface and semantics as [`cuda::std::barrier`], with the following additional operations

I think this section needs to be moved to the top, and it has to be clarified that `cuda::barrier` has the same interface but not the same semantics as `cuda::std::barrier`.  

It needs to say that, 
- if `!(scope == thread_block_scope && __isShared(this))`, then the semantics are the same as `cuda::std::barrier`, 
- otherwise, the semantics are that of "modify [....] of the c++ standard as follows:".